### DISCUSSION TITLE
Acessar dados de uma lista dentro de uma entidade

### DISCUSSION THREAD

**Discussion by wezmoreira** [balta-io/forum#183] [Tira Dúvidas] [answered]
Eu tenho uma entidade **"Order"** que dentro dela tem uma List<"Item"> dentro de **"Item"** eu tenho as propriedades **"Skuid e Name"** por exemplo. eu teria como acessar os dados dentro dessa lista usando algum comando como o "ForEachAsync()"?

Por exemplo:
```
    [HttpPost("v1/order")]
    public async Task<IActionResult> CreateOrderAsync(
        [FromServices] PottencialDataContext context,
        [FromBody] CreateOrderViewModel model)
    {
        var newOrder = new Order
        {
            Date = model.Date,
            Status = model.Status,
            SellerId = model.SellerId,
            Items = context.Items.ForEachAsync() // e acessar as propriedades
        };
    }

```
Dentro desse Post estou tentando acessar o "Skuid" que eu recebi no body da requisição e buscar ele no banco de dados, porem nao consigo acessar as propriedades dentro dessa lista.

Minha segunda duvida é, como eu adiciono multiplos valores? Por exemplo recebi 3 itens e tenho que adicionar eles, seria com o "AddRange()"?

tou tentando fazer isso pra recuperar a lista de Items:

```

        var listItem = new List<Item>();

        foreach (var item in model.Items)
        {
            var a = context.Items.FirstOrDefault(x => x.Skuid == item.Skuid);
            listItem.Add(a);
        }

```

e adicionar na Order:

```

        var newOrder = new Order
        {
            Date = model.Date,
            Status = model.Status,
            SellerId = model.SellerId,
            Items = listItem
        };


```

**Comment by andrebaltieri** [comment, marked as answer]
Boa noite, @wezmoreira, se não me engano seria assim 👇
```csharp
var a = var item = context.Orders.SelectMany(x => x.Items.Where(y => y.Skuid == "")).FirstOrDefault();
```
